docs(react-scheduler): add the scheduler Appointment Tooltip guide (#2054)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/guides/appointment-tooltip.md

# React Scheduler - Appointment Tooltip

> The React Scheduler component's `AppointmentTooltip` plugin displays compact appointment data.

Reviewer: I think it's better to replace 'compact' with 'detailed' and to mention that an appointment tooltip can contain some action buttons

Author: `Detailed` means `additional information`. `Compact` means `little information`. I think we should keep this word.

Reviewer: An appointment tooltip as a concept is often a panel with summarized info and actions - a date/time, description, resource name, and so on. From the other hand, appointments are often kept as minimalistic as possible to not distract an attention, e.g. in month view we render only a caption not date/time. So we can say that a tooltip displays extended/additional information :)

Author: Okay, you are right)

> An appointment tooltip displays brief information about the appointment and contains buttons that manage the appointment.

Reviewer 2: ```suggestion
An appointment tooltip can display information about the appointment and contains buttons that manage the appointment.
```

Reviewer 3: ```suggestion
An appointment tooltip displays information about the appointment and contains buttons that manage the appointment.
```

Reviewer 2: Ok

## Related Plugins

> The following plugins display an appointment data:

Reviewer: Appointment data -> appointment tooltip ?

> The following plugins are required:

Reviewer 2: Why 'plugins'? I can only see one here.

Reviewer 3: ```suggestion
The following plugin is required:
```

Reviewer 2: Ok

- [AppointmentTooltip](../reference/appointment-tooltip.md) - displays the appointment tooltip

## Basic Usage

> Import the plugin listed above. A user will be able to invoke the appointment tooltip by a click on an appointment.

Reviewer 2: ```suggestion
Import the plugin listed above. A user can click on an appointment to show its tooltip.
```

Reviewer 3: Ok

### Uncontrolled Mode

> The following demo shows the use of the appointment tooltip in uncontrolled mode:

Reviewer 2: Why 'the use of the'?

Reviewer 3: ```suggestion
The following demo shows how to use the appointment tooltip in uncontrolled mode:
```

Reviewer 2: Ok

.embedded-demo({ "path": "scheduler-tooltip/uncontrolled", "showThemeSelector": true })

### Controlled Mode

> In controlled mode, the state of all or particular `AppointmentTooltip`s should be controlled externally. To control their visibility, pass the `visible` property and `onVisibilityChange` event to the plugin. To control the displayed data, pass the [appointmentMeta](../reference/appointment-tooltip.md#appointmentmeta) property and `onAppointmentMetaChange` event.

Reviewer 2: 'all or particular'?

Reviewer 2: ```suggestion
In controlled mode, the state of `AppointmentTooltip`s should be controlled externally. To control their visibility, pass the `visible` property and `onVisibilityChange` event to the plugin. To manage the displayed data, pass the [appointmentMeta](../reference/appointment-tooltip.md#appointmentmeta) property and `onAppointmentMetaChange` event.
```

Reviewer 3: Ok

> In controlled mode, you should control the all/particular `AppointmentTooltip's` state. Pass the `visible` property and `onVisibilityChange`  event to control the visibility externally. Pass the [appointmentMeta](../reference/appointment-tooltip.md#appointmentmeta) property and `onAppointmentMetaChange` event to control the display data externally.

Reviewer: What does all/particular mean?

Author: The component has 2 state properties, you can change them both and one by one.

.embedded-demo({ "path": "scheduler-tooltip/controlled", "showThemeSelector": true })

### Customization

> To override default markup or add your own, you should define a `xxxComponent` properties, all list of these properties is available at plugin [API reference page](../reference/appointment-tooltip.md). In the following demo shows how to use `AppointmentTooltip's` `headerComponent` and `contentComponent` properties to make custom markup.

Reviewer: all list of these properties > a list of these properties

Reviewer: In the following demo shows -> The following demo shows

> To customize the appointment tooltip's appearance, redefine `xxxComponent` properties. `xxx` stands for the name of the element that you want to customize. For example, the tooltip's header and content are customized using the `headerComponent` and `contentComponent` properties, as shown in the code below. Refer to the [AppointmentTooltip API reference page](../reference/appointment-tooltip.md) for a list of all `xxxComponent` properties.

Reviewer 2: ```suggestion
To customize the tooltip's appearance, specify the `xxxComponent` properties (`xxx` is the name of the element you want to customize). For example, the `headerComponent` and `contentComponent` properties are used to customize the tooltip's header and content, as shown in the code below. Refer to the [AppointmentTooltip API reference page](../reference/appointment-tooltip.md) for a list of all `xxxComponent` properties.
```

Reviewer 3: Ok
